sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorBuilder.java

```diff
@@ -44,19 +62,141 @@
 
     private final ClientLogger logger = new ClientLogger(EventProcessorBuilder.class);
 
-    private Supplier<PartitionProcessor> partitionProcessorFactory;
+    private final EventHubClientBuilder eventHubClientBuilder;
     private String consumerGroup;
-    private PartitionManager partitionManager;
-    private EventHubAsyncClient eventHubAsyncClient;
+    private EventProcessorStore eventProcessorStore;
+    private Function<PartitionEvent, Mono<Void>> processEvent;
+    private Function<EventProcessingErrorContext, Mono<Void>> processError;
+    private Function<InitializationContext, Mono<Void>> initializePartition;
+    private Function<CloseContext, Mono<Void>> closePartition;
+
+    public EventProcessorBuilder() {
+        eventHubClientBuilder = new EventHubClientBuilder();
+    }
 
     /**
-     * Sets the Event Hub client the {@link EventProcessor} will use to connect to the Event Hub for consuming events.
+     * Sets the credential information given a connection string to the Event Hub instance.
      *
-     * @param eventHubAsyncClient The Event Hub asynchronous client for consuming events.
-     * @return The updated {@link EventProcessorBuilder} instance.
+     * <p>
+     * If the connection string is copied from the Event Hubs namespace, it will likely not contain the name to the
+     * desired Event Hub, which is needed. In this case, the name can be added manually by adding {@literal
+     * "EntityPath=EVENT_HUB_NAME"} to the end of the connection string. For example, "EntityPath=telemetry-hub".
+     * </p>
+     *
+     * <p>
+     * If you have defined a shared access policy directly on the Event Hub itself, then copying the connection string
+     * from that Event Hub will result in a connection string that contains the name.
+     * </p>
+     *
+     * @param connectionString The connection string to use for connecting to the Event Hub instance. It is expected
+     * that the Event Hub name and the shared access key properties are contained in this connection string.
+     * @return The updated {@link EventProcessorBuilder} object.
+     * @throws IllegalArgumentException if {@code connectionString} is null or empty. Or, the {@code connectionString}
```

Comment:
Is there a reason why this overload won't be throwing a `NullPointerException` but the next one does?

Reply:
Documentation error. Both throw NPE if connection string is null. Fixed documentation.